### PR TITLE
Support for TCP and TLS connections on service graph

### PR DESCRIPTION
**Pull request by liujed**
Introduced `akinet.TCPPacketMetadata`, `akinet.TCPConnectionMetadata`, and `api_schema.TCPConnectionReport`. Each TCP packet captured by the CLI produces packet metadata. These are gathered and summarized into connection-level metadata before being sent to the back end as TCP-connection reports.

During packet capture, TLS 1.2 and 1.3 Client Hello, Server Hello, and server certificate messages are parsed. From the TLS client, we extract the hostname from SNI and a list of protocols from ALPN. From the TLS server, we extract the TLS version. For TLS 1.2, we also extract the selected protocol from ALPN and the server's hostnames from the certificate's SAN extension. All this is encapsulated in `akinet.TLSClientHello` and `akinet.TLSServerHello`, which are gathered and summarized into `akinet.TLSHandshakeMetadata` before being sent to the back end as `api_schema.TLSHandshakeReport` messages.

Renamed `api_schema.GraphEdge` to `api_schema.HTTPGraphEdge`. Introduced `api_schema.TCPGraphEdge` and `api_schema.TLSGraphEdge`. Added TCP and TLS edges to `api_schema.GraphResponse`.

Extended `MemView` and `MemViewReader` to better support TLS parsing.

### CODE REVIEW

**Review by mgritter — APPROVED** [review]
Approved, but I could use some more explanation about how TCPConnectionMetadata could be "DestToSource"?

File: akinet/net_traffic.go [hunk 1]
```diff
@@ -44,6 +45,61 @@ func (rb RawBytes) String() string {
...
+	SYN bool
+
+	// Whether the ACK flag was set in the observed packet.
+	ACK bool
+
+	// Whether the FIN flag was set in the observed packet.
+	FIN bool
+
+	// Whether the RST flag was set in the observed packet.
+	RST bool
```
Comment (mgritter) [inline akinet/net_traffic.go:63]: I wonder whether we are going to be happier in the long run if we just mirror all the TCP flag bits as a single field.

🤷  This seems fine, we can change it later.

File: akinet/net_traffic.go [hunk 2]
```diff
@@ -44,6 +45,61 @@ func (rb RawBytes) String() string {
...
+}
+
+func (TCPConnectionMetadata) implParsedNetworkContent() {}
+
+type TCPConnectionDirection int
+
+const (
+	UnknownTCPConnectionDirection TCPConnectionDirection = iota
+	SourceToDest
+	DestToSource
```
[the remark this answers is not in the document]
Comment (liujed) [reply akinet/net_traffic.go:87]: Yes, that's exactly right. Happy to rename these if you can think of anything better.

File: akinet/net_traffic.go (hunk 1 above)
Comment (liujed) [reply akinet/net_traffic.go:63]: Yeah, not sure. They're already broken out into separate fields by the time they reach us.

File: akinet/net_traffic.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mgritter) [reply akinet/net_traffic.go:87]: Maybe SourceSentSYN/DestSentSYN?  SourceIsInitiator/DestIsInitiator?  I'm not in love with those either.

Or, perhaps a comment explaining that "direction" is "which of the two endpoints initiated the connection"

[the remark this answers is not in the document]
Comment (liujed) [reply akinet/net_traffic.go]: Renamed & documented in 0f624c8.

**Review by mgritter — APPROVED** [review]

**Pull request merged**